lib/Polaris.Class.ps1

```diff
@@ -0,0 +1,321 @@
+class Polaris {
+    
+    [Int]$Port
+    [System.Collections.Generic.List[PolarisMiddleWare]]$RouteMiddleWare = [System.Collections.Generic.List[PolarisMiddleWare]]::new()
+    [System.Collections.Generic.Dictionary[string, [System.Collections.Generic.Dictionary[string, string]]]]$ScriptBlockRoutes = [System.Collections.Generic.Dictionary[string, [System.Collections.Generic.Dictionary[string, string]]]]::new()
+    hidden [Action[string]]$Logger
+    hidden [System.Net.HttpListener]$Listener
+    hidden [System.Management.Automation.Runspaces.RunspacePool]$PowerShellPool = [RunspaceFactory]::CreateRunspacePool()
+    hidden [bool]$StopServer = $False
+    [string]$getLogsString = "PolarisLogs"
+
+
+    [Void] AddRoute(
+        [string]$path,
+        [string]$method,
+        [string]$scriptBlock
+    ) {
+        if ($scriptBlock -eq $null) {
+            throw [ArgumentNullException]::new("scriptBlock")
+        }
+
+        [string]$sanitizedPath = $path.TrimEnd('/').TrimStart('/')
+
+        if (-not $this.ScriptBlockRoutes.ContainsKey($sanitizedPath)) {
+            $this.ScriptBlockRoutes[$sanitizedPath] = [System.Collections.Generic.Dictionary[string, string]]::new()
+        }
+        $this.ScriptBlockRoutes[$sanitizedPath][$method] = $scriptBlock
+    }
+        
+    RemoveRoute(
+        [string]$path, 
+        [string]$method
+    ) {
+        if ($path -eq $null) {
+            throw [ArgumentNullException]::new("path")
+        }
+        if ($method -eq $null) {
+            throw [ArgumentNullException]::new("method")
+        }
+
+        [string]$sanitizedPath = $path.TrimEnd('/').TrimStart('/')
+        $this.ScriptBlockRoutes[$sanitizedPath].Remove($method)
+        if ($this.ScriptBlockRoutes[$sanitizedPath].Count -eq 0) {
+            $this.ScriptBlockRoutes.Remove($sanitizedPath)
+        }
+    }
+
+    AddMiddleware(
+        [string]$name,
+        [string]$scriptBlock
+    ) {
+        if ($scriptBlock -eq $null) {
+            throw [ArgumentNullException]::new("scriptBlock")
+        }
+        $this.RouteMiddleWare.Add
+        $this.RouteMiddleware.Add([PolarisMiddleware]@{
+                'Name'        = $name
+                'ScriptBlock' = $scriptBlock
+            })
+    }
+
+    RemoveMiddleware([string]$name) {
+        if ($name -eq $null) {
+            throw [ArgumentNullException]::new("name")
+        }
+        $this.RouteMiddleware.RemoveAll(
+            [Predicate[PolarisMiddleWare]]([scriptblock]::Create("`$args[0].Name -eq '$name'"))
+        )
+    }
+
+    [Void] Start(
+        [int]$port = 3000,  
+        [int]$minRunspaces = 1,
+        [int]$maxRunspaces = 1
+    ) {
+        $this.StopServer = $false
+
+        $this.PowerShellPool = [RunspaceFactory]::CreateRunspacePool($minRunspaces, $maxRunspaces)
+        $this.PowerShellPool.Open()
+        $this.Listener = $this.InitListener($port)
+
+        $syncHash = [hashtable]::Synchronized(@{})
```

Comment:
Some comments here would be really helpful for future contributors.

Reply:
Sorry I'm running short on time here but I pointed towards some blogs that explain how synchronized hashtables work.